openapi-cli/src/main/java/org/ballerinalang/ballerina/service/OpenApiResourceMapper.java

```diff
@@ -431,15 +417,9 @@ private void setArrayProperty(RequiredParameterNode queryParam, Map<String, Mode
             refModel.setReference(symbol.name().trim());
             ((RefProperty) symbolProperty).set$ref(refModel.get$ref());
 
-            Optional<Symbol> recordSymbol = semanticModel.symbol(queryParam.syntaxTree().filePath(),
-                    LinePosition.from(field.location().lineRange().startLine().line(),
-                            field.location().lineRange().startLine().offset()));
             //Set the record model to the definition
-            VariableSymbol recordVariable = (VariableSymbol) recordSymbol.orElseThrow();
-            ArrayTypeSymbol arrayTypeSymbol = (ArrayTypeSymbol) recordVariable.typeDescriptor();
-            if (arrayTypeSymbol.memberTypeDescriptor() instanceof TypeReferenceTypeSymbol) {
-                TypeReferenceTypeSymbol typeRecord =
-                        (TypeReferenceTypeSymbol) arrayTypeSymbol.memberTypeDescriptor();
+            if (symbol instanceof TypeReferenceTypeSymbol) {
```

Comment:
Consider using `type.kind()` for the validations, instead of using `instanceof` checks.